<?xml version="1.0"?>
<launch>

    <node pkg="exercise6" type="find_plane" name="find_plane" output="screen">

        <remap from="input" to="/camera/depth_registered/points"/>
        <remap from="/exercise6/find_plane" to="plane"/>

    </node>
    
        
    <node pkg="exercise6" type="cloud_voxelgrid" name="voxelgrid" output="screen">

        <!--remap from="input" to="/camera/depth_registered/points"/-->
        <remap from="input" to="/plane"/>
        <remap from="output" to="/exercise6/voxelgrid"/>

    </node>
    

    <node pkg="exercise6" type="find_blu" name="find_blu" output="screen">

        <remap from="input" to="/exercise6/voxelgrid"/>
        <remap from="filteredBlue" to="/exercise6/find_blu"/>

    </node>
     <node pkg="exercise6" type="find_grn" name="find_grn" output="screen">

        <remap from="input" to="/exercise6/voxelgrid"/>
        <remap from="filteredGreen" to="/exercise6/find_grn"/>

    </node>
     <node pkg="exercise6" type="find_red" name="find_red" output="screen">

        <remap from="input" to="/exercise6/voxelgrid"/>
        <remap from="filteredRed" to="/exercise6/find_red"/>

	</node>     
	<node pkg="exercise6" type="find_cyl" name="find_cyl" output="screen">

        <remap from="input" to="/exercise6/voxelgrid"/>
        <remap from="cylinder" to="/exercise6/find_cyl"/>

    </node>	<node pkg="exercise6" type="seperate" name="seperate" output="screen">

        <remap from="input" to="/exercise6/voxelgrid"/>

    </node>


</launch>
